packages/server-wallet/src/__chain-test__/directly-funded-channel.test.ts

```diff
@@ -100,181 +135,88 @@ beforeAll(async () => {
-  expect(BN.sub(bBalanceFinal, bBalanceInit)).toEqual(bFunding);
-
-  // TODO: remove this
-  // The reason for the wait:
-  // - B CloseChannel objective succeeds BEFORE AssetTransferred event arrives
-  // - B has no funds in the channel, so B does not wait for an AssetTransferred event to complete the CloseObjective
-  // - AssetTransferred event arrives to B after the test finishes running
-  // - We see "Error: Unable to acquire a connection"
-  // - The error does NOT fail the test
-  await new Promise(r => setTimeout(r, 500));
-}, 50_000);
+const testCases: Array<LatencyOptions & {closer: 'A' | 'B'}> = [
+  {
+    dropRate: 0,
+    meanDelay: undefined,
+    closer: 'A',
+  },
+  {
+    dropRate: 0,
+    meanDelay: undefined,
+    closer: 'B',
+  },
+  {dropRate: 0.1, meanDelay: 50, closer: 'A'},
+  {dropRate: 0.1, meanDelay: 50, closer: 'B'},
+];
+test.each(testCases)(
+  `can successfully fund and defund a channel between two wallets with options %o`,
+  async options => {
+    TestMessageService.setLatencyOptions({a, b}, options);
+    const participantA: Participant = {
+      signingAddress: await aEngine.getSigningAddress(),
+      participantId: 'a',
+      destination: makeDestination(aAddress),
+    };
+    const participantB: Participant = {
+      signingAddress: await bEngine.getSigningAddress(),
+      participantId: 'b',
+      destination: makeDestination(bAddress),
+    };
+
+    const allocation: Allocation = {
+      allocationItems: [
+        {
+          destination: participantA.destination,
+          amount: aFunding,
+        },
+        {
+          destination: participantB.destination,
+          amount: bFunding,
+        },
+      ],
+      assetHolderAddress: ethAssetHolderAddress,
+    };
+
+    const channelParams: CreateChannelParams = {
+      participants: [participantA, participantB],
+      allocations: [allocation],
+      appDefinition: ethers.constants.AddressZero,
+      appData: constants.HashZero,
+      fundingStrategy: 'Direct',
+      challengeDuration: ONE_DAY,
+    };
+    const aBalanceInit = await getBalance(aAddress);
+    const bBalanceInit = await getBalance(bAddress);
+    const assetHolderBalanceInit = await getBalance(ethAssetHolderAddress);
+
+    const response = await a.createChannels([channelParams]);
+    await waitForObjectiveProposals([response[0].objectiveId], b);
+    const bResponse = await b.approveObjectives([response[0].objectiveId]);
+
+    await expect(response).toBeObjectiveDoneType('Success');
+    await expect(bResponse).toBeObjectiveDoneType('Success');
+
+    const assetHolderBalanceUpdated = await getBalance(ethAssetHolderAddress);
+    expect(BN.sub(assetHolderBalanceUpdated, assetHolderBalanceInit)).toEqual(
+      BN.add(aFunding, bFunding)
+    );
+
+    const {channelId} = response[0];
+    const closeResponse =
+      options.closer === 'A'
+        ? await a.closeChannels([channelId])
+        : await b.closeChannels([channelId]);
```

Comment:
For future consideration, I like the pattern where we have something like a `peers: { A: Wallet, B: Wallet, all: [Wallet, Wallet]}` object, such that we can do
```suggestion
    const closeResponse = peers[options.closer].closeChannels([channelId]);
```

or
```
await Promise.all(peers.all.map(w => w.destroy()));
```